### PR TITLE
Explicitly require a C++11 capable toolchain when building Kyua

### PR DESCRIPTION
**Pull request by ngie-eign**
While it's true that C++11 isn't strictly required with ATF/Kyua, in order to move forward and leverage new C++ language features, the project needs to require newer C++ standards.

This will also allow Kyua to shed backwards compatibility code for `std::shared_ptr`, etc, further streamlining the project.

### CODE REVIEW

**Comment by jmmv** [comment]
Have you seen `utils::text::regex`?

Not saying requiring C++ is a bad idea (in fact, I wanted to do that at some point and then clean up things like `utils/shared_ptr.hpp`), but I have to question why this would be a prerequisite for googletest support.

File: NEWS.md [hunk 1]
```diff
@@ -7,8 +7,9 @@ Changes in version 0.14
 
 **NOT RELEASED YET; STILL UNDER DEVELOPMENT.**
 
-* No changes recorded.
-
+* Explicitly require C++-11 language features when compiling Kyua to avoid
+  having to rely on Boost for features made available in C++-11, e.g.,
```
Comment (jmmv) [inline NEWS.md]: The Boost comment here feels spurious. Kyua doesn't use Boost anywhere, so this mention comes out of the blue.

Comment (jmmv) [inline NEWS.md]: The Boost comment here feels spurious. Kyua doesn't use Boost anywhere, so this mention comes out of the blue.

File: configure.ac [hunk 2]
```diff
@@ -58,6 +58,7 @@ KYUA_MEMORY
 AC_CHECK_FUNCS([putenv setenv unsetenv])
 AC_CHECK_HEADERS([termios.h])
 
+AX_CXX_COMPILE_STDCXX(11, noext, mandatory)
```
Comment (jmmv) [inline configure.ac]: The rest of this file is careful to quote all arguments within `[]`. Please do the same.

Also this line checks features of the C++ compiler, like the block just above, so remove the blank line before this maybe.

**Review by jmmv — COMMENTED** [review]
Do we really need three commits to add this? Looks like a single one would be sufficient...

**Comment by ngie-eign** [comment]
> Have you seen `utils::text::regex`?

I hadn't seen that. I was trying to stick to what was provided upstream and I completely missed that... hmmm.

> Not saying requiring C++ is a bad idea (in fact, I wanted to do that at some point and then clean up things like `utils/shared_ptr.hpp`), but I have to question why this would be a prerequisite for googletest support.

I was unaware that the regex support had already been implemented using the C standard libraries, instead of the C++ standard libraries.

Given the number of bugs in FreeBSD's regex parsing logic (in the past), I think it would be better to settle on a C++ standard (since it's probably more standard across different implementations than the C ones).

I can work towards removing unnecessary (post-C++-11) code in another PR.

**Comment by ngie-eign** [comment]
> Do we really need three commits to add this? Looks like a single one would be sufficient...

I was trying to break down the work in terms of individual commits (out of habit), but I can squash them together.

**Comment by jmmv** [comment]
Breaking stuff into individual, smaller commits, is great -- but stuff should be grouped semantically. And in this case, the three modified files were changed to achieve the same thing. Thanks for merging them.

File: NEWS.md [hunk 3]
```diff
@@ -7,8 +7,7 @@ Changes in version 0.14
 
 **NOT RELEASED YET; STILL UNDER DEVELOPMENT.**
 
-* No changes recorded.
-
```
Comment (jmmv) [inline NEWS.md]: One last thing: you removed two lines but added one. Note that all other subsections are separate from each other by two blank lines, not one.

Also, the proper spelling is C++11, without the dash.

**Comment by jmmv** [comment]
By the way, for these little nits, I could just pull your commit, squash the minor edits myself, and then merge the change. I think this would mess the GitHub UI in that it wouldn't show the PR as merged, but if you are OK with it, I can do that to avoid roundtrips.

**Comment by ngie-eign** [comment]
> By the way, for these little nits, I could just pull your commit, squash the minor edits myself, and then merge the change. I think this would mess the GitHub UI in that it wouldn't show the PR as merged, but if you are OK with it, I can do that to avoid roundtrips.

I'm totally fine with that. It should show the merged commits as a joint effort with us both contributing :).

**Comment by ngie-eign** [comment]
@jmmv: could you please merge this change?

**Comment by ngie-eign** [comment]
Ugh, the `std::regex` implementation done by libc++ and libstdc++ are buggy, so the original reason for moving to C++11 is diminished a bit. However, it makes sense to move forward with this change.

**Comment by jmmv** [comment]
Yep, I think it makes sense to move forward too. It's 2019.

Interested in killing shared_ptr next and thus putting this to use? :)

**Comment by ngie-eign** [comment]
> Yep, I think it makes sense to move forward too. It's 2019.
> 
> Interested in killing shared_ptr next and thus putting this to use? :)

Will do :).

**Pull request merged**